api/src/main/java/org/openmrs/api/VisitService.java

```diff
@@ -295,6 +295,18 @@ public List<Visit> getVisits(Collection<VisitType> visitTypes, Collection<Patien
 	public List<Visit> getVisitsByPatient(Patient patient, boolean includeInactive, boolean includeVoided)
 	        throws APIException;
 	
+	/**
+	 * Gets all visits for the specified patient that started on or after the given start date
+	 * @param patient
+	 * @param maxStartDatetime
+	 * @param minEndDatetime
+	 * @return a list of visits
+	 * @throws APIException
+	 */
+	@Authorized(PrivilegeConstants.GET_VISITS)
+	public List<Visit> getVisitsByPatient(Patient patient, Date maxStartDatetime, Date minEndDatetime)
+	        throws APIException;
```

Comment:
I don't understand the search criteria or use case here, but I would much rather see us support a method like `getVisits(VisitSearchCriteria criteria)` where VisitSearchCriteria contains properties including `patient`, `minStartDatetime`, `maxStartDatetime`, `minEndDateTime`, `maxEndDateTime`, etc.  It seems odd to have a method that takes these very specific search critieria, and will be limiting in the future when we inevitably want to support limiting by other criteria.  Better to add additional possible, nullable attributes to `VisitSearchCriteria` that create lots of service methods with overlapping but slightly different arguments.

Reply:
That's a great suggestion. I created a method with search criteria here: https://github.com/openmrs/openmrs-core/pull/4709/commits/1b54547d89d7da85f7c051cf01e7c0d0a9df72cd

There's a set of overloaded methods for searching visits, I'll update those in a separate PR.